### PR TITLE
Create k6t periodic lanes for k8s-1.20 

### PR DESCRIPTION
**Pull request by qinqon**
This is part of checking k8s 1.20 lanes status and try to graduate to
github reporting.

It also add counterpart presubmit jobs with `always_run: false` to run them manually at `cluster: prow-workloads`

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Let's keep the `pull-kubevirt-e2e-k8s-1.20-cgroupsv2` lane executed on-demand.
```suggestion
    always_run: false
```

**Review by fgimenez — COMMENTED** [review]
/lgtm

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:9]: How this snicked in here? I do not think it is related to the sig-network run, right?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:32]: This PR enabled the sig-network job, not the storage one.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:51]: I think we want to test this first, right? (for a few days?)

**Review by EdDev — CHANGES_REQUESTED** [review]
Thanks. Please elaborate what is done exactly in the commit, it is hard to understand if the intention is in sync with the change itself.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:481]: I think you can drop `Multus` and `Macvtap` from this list.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:11]: typo

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:71]: 20

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:90]: true?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:130]: true?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:171]: true?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:11]: Please mention that we set this on the new cluster in the commit message.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (EdDev) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:33]: Same here, `Multus` and `Macvtap` are not needed.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:11]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

**Review by EdDev — APPROVED** [review]
Thanks!

**Comment by qinqon** [comment]
/hold cancel
@fgimenez can you approve ?

**Review by dhiller — COMMENTED** [review]
So, IIUC the general plan is to create new periodics to check the 1.20 lane at first, besides being able to work on stability in pull requests by manually triggering the jobs?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (oshoval) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:11]: New cluster will xfail the two tests of sig-network that use the docker bridge,

also it wont work still on the new cluster, (some labels should be changed) unless @fgimenez  already fixed it

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (oshoval) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:33]: see please k8s-1.19
need quotes and to use '\\[' ' \\]'

**Review by oshoval — CHANGES_REQUESTED** [review]

**Comment by oshoval** [comment]
PR title k6t -> k8s please

**Comment by qinqon** [comment]
> PR title k6t -> k8s please

k6t = kubevirt

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (oshoval) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:12]: we want skip report = true ?
optional = true is enough i think ?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:33]: This match the same tests.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:12]: yes we want them this way, while we test them on periodic.

**Comment by qinqon** [comment]
> New cluster will xfail the two tests of sig-network that use the docker bridge,
> (i see now they are skipped, but 2nd point here is valid)
> 
> also it wont work still on the new cluster, (some labels should be changed) unless @fgimenez already fixed it
> See please #1081 (comment)

@fgimenez is possible to run sig-network tests on the prow-workloads cluster ?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (oshoval) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:33]: We discussed in the past that we should prefect match including the brackets.
I think we should align it to the other jobs.

Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (oshoval) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:33]: It won't work without quotes, please add

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (oshoval) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:481]: we dont skip Operator even that it has its own lane ?

Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:481]: We do https://github.com/kubevirt/project-infra/blob/master/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L104

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (oshoval) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:481]: Right i missed the Operator on the periodic (not on the presubmit)

**Review by oshoval — APPROVED** [review]
Thanks
Try please to run the periodics / presubmits after its merged if possible
(worth to verify each one runs the expected amount of tests)

**Comment by oshoval** [comment]
Generally speaking, not due to this PR, i think we should align the style of the modules,
some use `[[sig-x]]` some use `@sig-x` or just `x`

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (fgimenez) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:11]: Not yet, will let you all know when ready.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:306]: This is the exact same time that the 1.19-sig-network periodic uses, could you pls shift it a bit so that we trigger the periodics separately and don't have the risk to DDoS ourselves? Same for the rest, I think they match the 1.19 periodics trigger times.

**Review by fgimenez — COMMENTED** [review]
Looks very good thanks! Just shifting a bit the trigger times (taking minutes and hours that are not used by other periodics) should be enough.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (oshoval) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:11]: Thanks, but meanwhile is it fine to `cluster: prow-workloads` and putting the 3 labels as false right ?
As this what was done in this PR
Following your tip here https://github.com/kubevirt/project-infra/pull/1081#issuecomment-813964129

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (qinqon) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml]: Done

**Comment by qinqon** [comment]
@fgimenez does it make sense to split yamls per k8s versions ? it will be easier to copy paste and modify and remove

so we will store this on `kubevirt-periodics-k8s-1.20.yaml` and `kubevirt-presubmit-k8s-1.20.yaml`

**Comment by oshoval** [comment]
In order to make it easier to review and find mistakes, i think its worth at least next time, to create one commit which copy an existing job, and then the other commits show the diff against it, this way we know exactly what was changed.

**Comment by fgimenez** [comment]
> @fgimenez does it make sense to split yamls per k8s versions ? it will be easier to copy paste and modify and remove
> 
> so we will store this on `kubevirt-periodics-k8s-1.20.yaml` and `kubevirt-presubmit-k8s-1.20.yaml`

Ok sounds good but maybe for a follow-up, it is a big change in structure but I agree that would help in the future. It would be good to ping people to ask their opinion about this change.

**Review by fgimenez — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
